recipe-client-addon/test/unit/test_PreferenceExperiments.js

```diff
@@ -165,6 +172,24 @@ add_task(withMockExperiments(withMockPreferences(async function (experiments, mo
   startObserver.restore();
 })));
 
+// start should detect if a new preference value type matches the previous value type
+add_task(withMockPreferences(async function (mockPreferences) {
+  mockPreferences.set("fake.type_preference", "oldvalue");
+
+  await Assert.rejects(
+    PreferenceExperiments.start({
+      name: "test",
+      branch: "branch",
+      preferenceName: "fake.type_preference",
+      preferenceBranchType: "invalid",
```

Comment:
You'll want to use a valid preference branch type (`user`, in this case), or else `start` will throw for an invalid branch type, not due to an invalid preference type.

It looks like [`Assert.throws` takes an `expected` argument](https://dxr.mozilla.org/mozilla-central/source/testing/modules/Assert.jsm#381) that we can use to assert that error message matches the cause for the exception that we expect. If you follow the code it looks like we can pass a regex object in to match against a portion of the error message. Someday we should refactor the rest of the throws tests to use this too, it looks useful!